awsbestpractices/loops-to-batches/LoopsToBatches.java

```diff
@@ -0,0 +1,185 @@
+package example;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.jms.Connection;
+import javax.jms.Destination;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+import com.amazonaws.services.codedeploy.AmazonCodeDeployClient;
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.activemq.command.ActiveMQQueue;
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+
+public class LoopsToBatches {
+    AmazonDynamoDB dynamoDBClient = AmazonDynamoDBClientBuilder.standard().build();
+
+    public void batchablePutItemOnCollection_1(JobExecutionContext context) throws JobExecutionException {
+        MessageConsumer consumer = session.createConsumer(queue);
+        for (int i = 1; i < numMsgs + 1; i++) {
+            Map<String, AttributeValue> item = new HashMap<String, AttributeValue>();
+            TextMessage txtMessage = (TextMessage) consumerMessage;
+            String msgBody = txtMessage.getText().toString();
+            item.put("msg_body", new AttributeValue().withS(msgBody));
+            // Can be batched
+            dynamoDBClient.putItem(getTableName, item);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is using a single API operation, but a batch API operation exists. You could help improve efficiency by using the batch operation instead. The batch operation is batchWriteItem.